app/controllers/users_controller.rb

```diff
@@ -0,0 +1,54 @@
+class UsersController < ApplicationController
+
+    get '/users/:slug' do 
+        @user = User.find_by_slug(params[:slug])
+        erb :'/users/show'
+    end
+
+    get '/signup' do 
+        if !logged_in?             
+            erb :'/users/create_user'
+        else
+            redirect to '/tweets'
+        end
+    end
+
+    post '/signup' do
+        if params[:username] == "" || params[:email] == "" || params[:password] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.